sale_product_set_packaging_qty/models/product_set.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_is_zero
+
+
+class ProductSetLine(models.Model):
+    _inherit = "product.set.line"
+
+    product_packaging_id = fields.Many2one(
+        "product.packaging", domain="[('product_id', '=', product_id)]"
+    )
+    product_packaging_qty = fields.Float(
+        compute="_compute_product_packaging_qty",
+        inverse="_inverse_product_packaging_qty",
+    )
+
+    @api.depends("quantity", "product_packaging_id", "product_packaging_id.qty")
+    def _compute_product_packaging_qty(self):
+        for line in self:
+            uom_rounding = line.product_id.uom_id.rounding
+            if (
+                not line.product_packaging_id
+                or float_is_zero(line.quantity, precision_rounding=uom_rounding)
+                or float_is_zero(
+                    line.product_packaging_id.qty, precision_rounding=uom_rounding
+                )
+            ):
+                line.product_packaging_qty = 0
+                continue
+            line.product_packaging_qty = line.quantity / line.product_packaging_id.qty
```

Comment:
If the packaging is not set we should not have any qty here. No matter the rest.
Right not the inverse method is broken on `update` because of this.

Use case:
* 1 line w/ no packaging -> pkg qty gets computed
* you call `update` to set packaging + qty -> you get an error

BTW the validation should be done in a constraint IMO.

Reply:
Hmm testing this a bit further it's quite weird: when using `update` by passing pkg qty and pkg, `product_packaging_qty` gets the new value immediately (hence the check on inverse is satisfied) but the pkg is not.
If you use write, both values are already updated when hitting the inverse method :thinking: